model_api/cpp/models/src/segmentation_model.cpp

```diff
@@ -208,73 +222,70 @@ std::unique_ptr<ResultBase> SegmentationModel::postprocess(InferenceResult& infR
+        cv::Mat predictions(outHeight, outWidth, CV_32SC1, outTensor.data<int32_t>());
+        predictions.convertTo(soft_prediction, CV_8UC1);
+    } else if (outChannels == 1 && outTensor.get_element_type() == ov::element::i64) {
+        cv::Mat predictions(outHeight, outWidth, CV_32SC1);
+        const auto data = outTensor.data<int64_t>();
+        for (size_t i = 0; i < predictions.total(); ++i) {
+            reinterpret_cast<int32_t*>(predictions.data)[i] = int32_t(data[i]);
         }
+        predictions.convertTo(soft_prediction, CV_8UC1);
+    } else if (outTensor.get_element_type() == ov::element::f32) {
+        float* data = outTensor.data<float>();
+        std::vector<cv::Mat> channels;
+        for (size_t c = 0; c < outTensor.get_shape()[1]; ++c) {
+            channels.emplace_back(cv::Size{outWidth, outHeight}, CV_32FC1, data + c * outHeight * outWidth);
+        }
+        cv::merge(channels, soft_prediction);
+    }
 
-        cv::resize(result->resultImage,
-                result->resultImage,
-                cv::Size(inputImgSize.inputImgWidth, inputImgSize.inputImgHeight),
-                0,
-                0,
-                cv::INTER_NEAREST);
+    cv::Mat hard_prediction = create_hard_prediction_from_soft_prediction(
+        soft_prediction, soft_threshold, blur_strength);
 
-        return std::unique_ptr<ResultBase>(result);
-    }
-    float* data = outTensor.data<float>();
-    std::vector<cv::Mat> channels;
-    for (size_t c = 0; c < outTensor.get_shape()[1]; ++c) {
-        channels.emplace_back(cv::Size{outHeight, outWidth}, CV_32F, data + c * outHeight * outWidth);
-    }
-    cv::Mat soft_prediction;
-    cv::merge(channels, soft_prediction);
-    cv::Mat hard_prediction = create_hard_prediction_from_soft_prediction(soft_prediction, soft_threshold, blur_strength);
     cv::resize(hard_prediction, hard_prediction, {inputImgSize.inputImgWidth, inputImgSize.inputImgHeight}, 0.0, 0.0, cv::INTER_NEAREST);
+
     if (return_soft_prediction) {
-        cv::resize(soft_prediction, soft_prediction, {inputImgSize.inputImgWidth, inputImgSize.inputImgHeight}, 0.0, 0.0, cv::INTER_NEAREST);
         ImageResultWithSoftPrediction* result = new ImageResultWithSoftPrediction(infResult.frameId, infResult.metaData);
         result->resultImage = hard_prediction;
+        cv::resize(soft_prediction, soft_prediction, {inputImgSize.inputImgWidth, inputImgSize.inputImgHeight}, 0.0, 0.0, cv::INTER_NEAREST);
         result->soft_prediction = soft_prediction;
         return std::unique_ptr<ResultBase>(result);
+    } else {
+        ImageResult* result = new ImageResult(infResult.frameId, infResult.metaData);
+        result->resultImage = hard_prediction;
+        return std::unique_ptr<ResultBase>(result);
     }
-    ImageResult* result = new ImageResult(infResult.frameId, infResult.metaData);
-    result->resultImage = hard_prediction;
-    return std::unique_ptr<ResultBase>(result);
```

Comment:
You don't need `else` if `if` ends with `return`. Revert this please

Reply:
I didn't know that there are people who prefer that style.

The problem with changing style in scope of other work is that a reviewer may not like the change or simply be unaware of other style. In that context such changes cause questions.

I'm not saying that you shouldn't modify style in a PR at all. That's very convenient to fix something minor or stylish while you have a file opened. I'm saying that, that the bigger piece of code you modify the more reasons a reviewer has to ask questions/start an argument.